shared/util/tf/tf_generator.go

```diff
@@ -0,0 +1,439 @@
+package tf
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/chanzuckerberg/happy/shared/config"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/gocty"
+)
+
+type provider struct {
+	Name    string
+	Source  string
+	Version string
+}
+
+type variable struct {
+	Name        string
+	Type        string
+	Description string
+	Default     cty.Value
+}
+
+var requiredProviders []provider = []provider{
+	{
+		Name:    "aws",
+		Source:  "hashicorp/aws",
+		Version: ">= 4.45",
+	},
+	{
+		Name:    "kubernetes",
+		Source:  "hashicorp/kubernetes",
+		Version: ">= 2.16",
+	},
+	{
+		Name:    "datadog",
+		Source:  "datadog/datadog",
+		Version: ">= 3.20.0",
+	},
+	{
+		Name:    "happy",
+		Source:  "chanzuckerberg/happy",
+		Version: ">= 0.53.5",
+	},
+}
+
+var requiredVariables []variable = []variable{
+	{
+		Name:        "aws_account_id",
+		Type:        "string",
+		Description: "AWS account ID to apply changes to",
+	},
+	{
+		Name:        "k8s_cluster_id",
+		Type:        "string",
+		Description: "EKS K8S Cluster ID",
+	},
+	{
+		Name:        "k8s_namespace",
+		Type:        "string",
+		Description: "K8S namespace for this stack",
+	},
+	{
+		Name:        "aws_role",
+		Type:        "string",
+		Description: "Name of the AWS role to assume to apply changes",
+	},
+	{
+		Name:        "image_tag",
+		Type:        "string",
+		Description: "Please provide an image tag",
+	},
+	{
+		Name:        "image_tags",
+		Type:        "string",
+		Description: "Override the default image tags (json-encoded map)",
+		Default:     cty.StringVal("{}"),
+	},
+	{
+		Name:        "stack_name",
+		Type:        "string",
+		Description: "Happy Path stack name",
+	},
+	{
+		Name:        "wait_for_steady_state",
+		Type:        "bool",
+		Description: "Should terraform block until k8s deployment reaches a steady state?",
+		Default:     cty.BoolVal(true),
+	},
```

Comment:
```suggestion
	
```

Reply:
This variable is present in `happy_env_ecs` module, I don't think we can drop it.